src/lib/protocols/resp.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * resp.c
+ *
+ * Redis Serialization Protocol
+ * 
+ * Copyright (C) 2024 - ntop.org
+ * Copyright (C) 2024 - V.G <jacendi@protonmail.com>
+ *
+ * This file is part of nDPI, an open source deep packet inspection
+ * library based on the OpenDPI and PACE technology by ipoque GmbH
+ *
+ * nDPI is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * nDPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with nDPI.  If not, see <http://www.gnu.org/licenses/>.
+ * 
+ */
+
+#include "ndpi_protocol_ids.h"
+
+#define NDPI_CURRENT_PROTO NDPI_PROTOCOL_RESP
+
+#include "ndpi_api.h"
+#include "ndpi_private.h"
+
+static void ndpi_search_resp(struct ndpi_detection_module_struct *ndpi_struct,
+                             struct ndpi_flow_struct *flow)
+{
+  struct ndpi_packet_struct const * const packet = &ndpi_struct->packet;
+
+  NDPI_LOG_DBG(ndpi_struct, "search RESP\n");
+  
+  if (packet->payload_packet_len < 10)
+    goto exclude;
+
+  switch(packet->payload[0])
+  {
+    case '*':
+    case '$':
+    case '~':
+      break;
+    default:
+      goto exclude;
+  }
+
+  u_int8_t offset = 1;
+  while (offset < 4 && packet->payload[offset] != '\r')
+  {
+    if (!isascii(packet->payload[offset]))
+      goto exclude;
+    offset++;
+  }
+
+  if (memcmp(&packet->payload[offset], "\r\n", 2) == 0) {
+    NDPI_LOG_INFO(ndpi_struct, "found RESP\n");
+    ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_RESP,
+                               NDPI_PROTOCOL_UNKNOWN, NDPI_CONFIDENCE_DPI);
+  }
```

Comment:
@IvanNardi @utoni what do you think about that? Could it be optimized in some way?

Reply:
`ndpi_isdigit` instead of `isascii`? [if we are looking for a number...]